system-x/services/splunk/src/main/java/software/tnb/splunk/service/openshift/Splunk.java

```diff
@@ -0,0 +1,220 @@
+
+import cz.xtf.core.openshift.helpers.ResourceParsers;
+import io.fabric8.kubernetes.api.model.DeletionPropagation;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.IntOrString;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.rbac.ClusterRole;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import io.fabric8.openshift.api.model.Route;
+import io.fabric8.openshift.api.model.RouteBuilder;
+
+@AutoService(Splunk.class)
+public class Splunk implements Service, OpenshiftDeployable, WithExternalHostname {
+    private static final Logger LOG = LoggerFactory.getLogger(Splunk.class);
+    public static final String ROUTE_NAME = "api";
+    public static final int API_PORT = 443;
+    private SplunkAccount account;
+    private com.splunk.Service client;
+    private SplunkValidation validation;
+    private List<HasMetadata> createdResources;
+    private CustomResourceDefinitionContext crdContext;
+    private Route apiRoute;
+
+    private static final String SCC_NAME = "tnb-splunk";
+
+    public SplunkAccount account() {
+        if (account == null) {
+            LOG.debug("Creating new Splunk account");
+            account = AccountFactory.create(SplunkAccount.class);
+            account.setPassword(
+                new String(Base64.getDecoder().decode(OpenshiftClient.get().getSecret("splunk-s1-standalone-secret-v1").getData().get("password"))));
+        }
+        return account;
+    }
+
+    /**
+     * Due to self sign certificate, the client is not able to communicate via localhost and port-forward.
+     * OCP external route with `reencrypt` is used. (Cluster needs to have valid certificate!)
+     */
+    protected com.splunk.Service client() {
+        LOG.debug("Creating new Splunk client");
+        ServiceArgs loginArgs = new ServiceArgs();
+        loginArgs.setUsername(account().getUsername());
+        loginArgs.setPassword(account().getPassword());
+        loginArgs.setHost(this.externalHostname());
+        loginArgs.setPort(API_PORT);
+        loginArgs.setScheme("https");
+
+        client = com.splunk.Service.connect(loginArgs);
+        return client;
+    }
+
+    public SplunkValidation validation() {
+        return validation;
+    }
+
+    @Override
+    public void create() {
+        OpenshiftClient.get().addUsersToSecurityContext(
+            OpenshiftClient.get().createSecurityContext(SCC_NAME, "nonroot"),
+            "system:serviceaccount:" + OpenshiftConfiguration.openshiftNamespace() + ":splunk-operator-controller-manager",
+            "system:serviceaccount:" + OpenshiftConfiguration.openshiftNamespace() + ":default");
+
+        try {
+            if (getSplunkCrd() == null) { // test if CRD already exists on cluster
+                LOG.info("Creating Splunk CRD's from file {}", "splunk-crds.yaml");
```

Comment:
you can include the name directly in the log if you don't have it in any variable :grin: